src/Doctrine/ManagerRegistry.php

```diff
@@ -29,7 +29,12 @@ class ManagerRegistry implements ManagerRegistryInterface
     /**
      * @var EntityManager[]
      */
-    protected $managers;
+    protected $originalManagers;
+
+    /**
+     * @var EntityManager[]
+     */
+    protected $resetManagers = [];
```

Comment:
Hi there,

The previous $mangers member was not an array as the docblock suggests but a container (because `$this->managers = $this->container['orm.ems'];` and the `DoctrineOrmServiceProvider` defines `'orm.ems'` as being `$ems = new Container(); ... return $ems`). Since container services are frozen once they get instantiated once, it means that I was not able to replace the old service with a new one.

Making the previous $manages member be an actual array instead of a container was not a solution because that would have instantiated all the EMs from the `'orm.ems'` container which defeats the purpose of a container.

So my solution was to leave the old $managers member as a container and rename it to $originalManagers and add the $resetManagers and when serving an EM I would pull from these giving priority to the $resetManagers member.